tbx/blog/models.py

```diff
@@ -196,9 +190,28 @@ def related_blog_posts(self):
             )
             .distinct()
             .order_by("-date")
-            .exclude(pk=self.pk)[:3]
+            .exclude(pk=self.pk)
         )
 
+        if final_division := getattr(self, "final_division", None):
+            # Ideally this would be implemented at the database-level but it's
+            # very hard to implement the final_division logic at the ORM level
+            def filter_fn(page):
+                return page.final_division == final_division
+
+            queryset = list(filter(filter_fn, queryset))
```

Comment:
(minor) I wonder if it would be possible or make sense to boost items with the same set of service tags.

Reply:
It should be possible by creating a custom sorting function but that would require 1-2h of extra development time.
I'd suggest creating a new ticket and clarify the requirements with regards to boosting:

- Do we consider only services or also sectors?
- Do matching services contribute more/less/equally to matching sectors?
- Does the sector/service list need to match exactly to be boosted?
- (in the case of partial matching in the previous point) Does a page with more matching sectors/services get boosted more

Reply:
Good point and great questions. Happy to split that into its own ticket